src/hmac.rs

```diff
@@ -72,10 +78,44 @@ where
 
     let mut okm = vec![0u8; K::PrivKey::LENGTH];
     hk.expand(info, &mut okm)
-        .map_err(|_| signature::Error::new())?;
+        .map_err(|_| FastCryptoError::GeneralError)?;
 
-    let secret_key = K::PrivKey::from_bytes(&okm[..]).map_err(|_| signature::Error::new())?;
+    let secret_key = K::PrivKey::from_bytes(&okm[..]).unwrap();
 
     let keypair = K::from(secret_key);
     Ok(keypair)
 }
+
+////////////////////////////////////////////////////////////////////////
+/// HMAC-SHA256 based functions
+
+const HMAC_KEY_RECOMMENDED_LENGTH: usize = 32;
+const HKDF_KEY_RECOMMENDED_LENGTH: usize = 16;
```

Comment:
why HKDF recommended with 16?

Reply:
128 bit security should be enough for hkdf

Reply:
I would prefer setting it to 256bits. Note that this is a generic HKDF api and it depends on the app context where it will be used, but if used for key-pair derivation in a mass scale (ie deriving millions of keys), the effective security is reduced due to multi-target attacks. This is unfortunately one of the reasons that NO blockchain today (due to mnemonics being short) offer in practice 128 bits. See https://twitter.com/kostascrypto/status/1565141220115881984 -> we're preparing a related short abstract to publish in IACR.

Reply:
done